corehq/apps/userreports/reports/view.py

```diff
@@ -250,7 +252,27 @@ def request_user(self):
     @property
     @memoized
     def filter_values(self):
-        return get_filter_values(self.filters, self.request_dict, user=self.request_user)
+        report_filters = get_filter_values(self.filters, self.request_dict, user=self.request_user)
+        # Removing location from the filters for the locations that are not assigned to the current user.
+        if (toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(self.domain)
+                and not self.request_user.has_permission(self.domain, 'access_all_locations')):
+            location_key = None
+            user_location_ids = self.request_user.get_location_ids(self.domain)
+            user_filtered_locations = []
+            for k, v in report_filters.items():
+                if 'computed_owner_location' in k:
+                    location_key = k
+                    user_filtered_locations = [choice for choice in v if choice.value in user_location_ids]
+            if location_key:
+                if user_filtered_locations:
+                    report_filters[location_key] = user_filtered_locations
+                else:
+                    # Case where user is assigned a new dynamic location that is not present in report filters
+                    # In this case, user should not see any data as this location will not be assigned
+                    # to the user
+                    empty_location = Choice(value=uuid.uuid4().hex, display=None)
+                    report_filters[location_key] = [empty_location]
```

Comment:
I think this is equivalent to below. It simplifies conditional branching (easier to read)
```suggestion
            if user_filtered_locations:
                report_filters[location_key] = user_filtered_locations
            elif location_key:
                # Case where user is assigned a new dynamic location that is not present in report filters
                # In this case, user should not see any data as this location will not be assigned
                # to the user
                empty_location = Choice(value=uuid.uuid4().hex, display=None)
                report_filters[location_key] = [empty_location]
```